### PR TITLE
[GitBot] Add Publish and Compare Commands

### PR DESCRIPTION
**Pull request by trulyronak**
This PR contains features needed for the gitbot. This edits one workspace, `ui`. The UI changes are to enable the apidocs site, that lets users access their documentation online.

Core Changes
- deployment workflow for apidocs site
- updates to the latest version of @useoptic/domain

### CODE REVIEW

File: workspaces/ui/src/contexts/SpecViewerContext.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import React, { useMemo } from 'react';
+import axios from 'axios';
```
Comment (devdoshi) [inline workspaces/ui/src/contexts/SpecViewerContext.js:2]: any reason to use axios instead of fetch?

Comment (trulyronak) [reply workspaces/ui/src/contexts/SpecViewerContext.js]: i was running into issues with `fetch` — but that was before we did a lot of fixing — just retried it and it all worked so I switched over

Comment (devdoshi) [reply workspaces/ui/src/contexts/SpecViewerContext.js]: awesome. Thank you!

**Comment by trulyronak** [comment]
ignore my bot doing strange things :P (its resolved now)

**Pull request merged**